### ISSUE TITLE
Large number of boolean operators in query causes KSQL to hang

### ISSUE THREAD

**Issue by mbrancato** [confluentinc/ksql#1784]
I recently updated from a manually built 5.0.0-beta2 docker image to 5.0.0. Upon doing so, when starting in headless mode with a queries file, KSQL only loads about 70 queries, then stops. I'm using a large number of queries, and most are `INSERT INTO`.

I'm referring to the printing of these messages on load:

> [2018-08-26 03:05:42,019] INFO Building AST for INSERT INTO stream SELECT ...

Rolling back to beta30 (cp-ksql-server image) seems to work again loading all queries. Any idea if some limit on queries was added? I tried using the official ksql-cli (yes, running in headless mode) and cp-ksql-server with the same result.

Sidenote - I tried to build a 5.0.0 docker image locally, but even at a build success, I can't see any built image.

**Comment by apurvam** [comment]
There is no limit on queries. What do you mean by 'stopping'? are there any error messages? Can you load ksql in interactive mode and then do `RUN SCRIPT` on the queries?

**Comment by mbrancato** [comment]
After having some more time to test this, it does appear to be a number of boolean operators issue, not a number of queries issue. It was getting hung at a query (example) for about an hour. Then it would hit a longer query that KSQL could never get past.

@apurvam  - There are no error messages. As I stated, I'm defining 'loading' as the "Building AST" message. And KSQL just stops printing anything. I never see the more common messages about rebalancing, etc as it executes the queries.

I've created a minimal queries file that seems to exhibit this behavior:
```
CREATE STREAM events (
field0 varchar,
id int )
WITH (KAFKA_TOPIC='events', VALUE_FORMAT='json');

CREATE STREAM test (
field0 varchar,
field1 varchar,
id varchar )
WITH ( KAFKA_TOPIC='test', VALUE_FORMAT='json');

INSERT INTO test SELECT
event_id,
'x_0' AS field1,
field0
FROM events WHERE ((id=1 OR id=2 OR id=3 OR id=4) AND (field0='0x10' OR field0='0x11' OR field0='0x12' OR field0='0x13' OR field0='0x14' OR field0='0x15' OR field0='0x16' OR field0='0x17' OR field0='0x18' OR field0='0x19' OR field0='0x1A' OR field0='0x1B' OR field0='0x1C' OR field0='0x1D' OR field0='0x1E' OR field0='0x1F' OR field0='0x20' OR field0='0x21' OR field0='0x22' OR field0='0x23' OR field0='0x24' OR field0='0x25' OR field0='0x26' OR field0='0x27' OR field0='0x28' OR field0='0x29' OR field0='0x2A' OR field0='0x2B' OR field0='0x2C' OR field0='0x2D' OR field0='0x2E' OR field0='0x2F' OR field0='0x30' OR field0='0x31' OR field0='0x32' OR field0='0x33' OR field0='0x34' OR field0='0x35' OR field0='0x36' OR field0='0x37' OR field0='0x38' OR field0='0x39' OR field0='0x3A' OR field0='0x3B' OR field0='0x3C' OR field0='0x3D' OR field0='0x3E' OR field0='0x3F'));
```

**Comment by apurvam** [comment]
@dguy can you triage this as the on call?

**Comment by dguy** [comment]
yes - this is definitely an issue. The creation of the streams app for the insert gets progressively slower as you add more and more `OR` conditions to the statement. I suspect that it may eventually succeed, but not for a long time.

**Comment by dguy** [comment]
```
"qtp1046009438-37" #37 prio=5 os_prio=31 tid=0x00007fb214fdf000 nid=0x6303 runnable [0x000070000a26e000]
   java.lang.Thread.State: RUNNABLE
	at java.lang.StringCoding.encode(StringCoding.java:364)
	at java.lang.String.getBytes(String.java:941)
	at io.airlift.slice.Slices.copiedBuffer(Slices.java:292)
	at io.airlift.slice.Slices.utf8Slice(Slices.java:297)
	at io.confluent.ksql.parser.tree.StringLiteral.<init>(StringLiteral.java:44)
	at io.confluent.ksql.parser.tree.StringLiteral.<init>(StringLiteral.java:33)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitStringLiteral(StatementRewriter.java:425)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitStringLiteral(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.StringLiteral.accept(StringLiteral.java:57)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitComparisonExpression(StatementRewriter.java:172)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitComparisonExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.ComparisonExpression.accept(ComparisonExpression.java:130)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:699)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:692)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:698)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.lambda$visitLogicalBinaryExpression$43(StatementRewriter.java:693)
	at io.confluent.ksql.parser.rewrite.StatementRewriter$$Lambda$375/1058776400.apply(Unknown Source)
	at java.util.Optional.map(Optional.java:215)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:689)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitLogicalBinaryExpression(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.LogicalBinaryExpression.accept(LogicalBinaryExpression.java:85)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitQuerySpecification(StatementRewriter.java:264)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitQuerySpecification(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.QuerySpecification.accept(QuerySpecification.java:138)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitQuery(StatementRewriter.java:208)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitQuery(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.Query.accept(Query.java:66)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitInsertInto(StatementRewriter.java:1047)
	at io.confluent.ksql.parser.rewrite.StatementRewriter.visitInsertInto(StatementRewriter.java:111)
	at io.confluent.ksql.parser.tree.InsertInto.accept(InsertInto.java:63)
	at io.confluent.ksql.parser.tree.DefaultAstVisitor.process(DefaultAstVisitor.java:25)
	at io.confluent.ksql.parser.rewrite.StatementRewriteForStruct.rewriteForStruct(StatementRewriteForStruct.java:46)
	at io.confluent.ksql.parser.KsqlParser.buildAst(KsqlParser.java:59)
	at io.confluent.ksql.KsqlEngine.getStatements(KsqlEngine.java:494)
	at io.confluent.ksql.rest.server.resources.KsqlResource.handleKsqlStatements(KsqlResource.java:167)
	at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:76)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$$Lambda$299/13063602.invoke(Unknown Source)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:148)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:191)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:200)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:103)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:493)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:415)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:104)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:277)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:272)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:268)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:316)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:298)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:268)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:289)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:256)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:703)
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:416)
	at org.glassfish.jersey.servlet.ServletContainer.serviceImpl(ServletContainer.java:409)
	at org.glassfish.jersey.servlet.ServletContainer.doFilter(ServletContainer.java:584)
	at org.glassfish.jersey.servlet.ServletContainer.doFilter(ServletContainer.java:525)
	at org.glassfish.jersey.servlet.ServletContainer.doFilter(ServletContainer.java:462)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1317)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1219)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:169)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:219)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:531)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:352)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:762)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:680)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by hjafarpour** [comment]
This was a bug in StatementRewriter because we were using `map/orElse` instead of `if` statement. The following PR is the fix: https://github.com/confluentinc/ksql/pull/1801

**Comment by miguno** [comment]
Closing this because https://github.com/confluentinc/ksql/pull/1801 should have fixed this.  Please reopen if needed, @mbrancato.

**Comment by mbrancato** [comment]
@hjafarpour or @miguno  - I don't see this #1801 PR merged into the 5.1.x branch or master. Is there a process to merge these later?

**Comment by apurvam** [comment]
It is merged into 5.0.x, and should have subsequently been merged to master automatically. 5.1.x was cut afterward, so it should be there too. 

Do you not see the commit, or are you just asking out of curiosity?

**Comment by mbrancato** [comment]
@apurvam I'm asking because it isn't in 5.1.x or master. Just trying to point this out politely as it might cause other regressions if it isn't isolated.

5.0.x:
https://github.com/confluentinc/ksql/blob/1cce8a898ea480d0c8ba7f7ac1e46d6b1cf9a3ad/ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java#L151-L164

5.1.x:
https://github.com/confluentinc/ksql/blob/2296ddf3c864c3fe6ef738b9b096eec2d586ca05/ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java#L146-L160

master:
https://github.com/confluentinc/ksql/blob/73a56a63b80fb333629b08123e97e17e013e9f84/ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java#L146-L160

**Comment by apurvam** [comment]
Thanks for the details @mbracanto. It may well have been missed. 

@hjafarpour can you check, since it was your commit that is missing?

**Comment by apurvam** [comment]
I did a bit more digging, and there definitely are merge commits which haven't made the desired changes onto master.

Here is the commit history on master at around the time (Aug 31) when #1801 was merged: https://github.com/confluentinc/ksql/commits/master?after=73a56a63b80fb333629b08123e97e17e013e9f84+559

You can definitely see that the specific commit was merged to master. But if you browse the tree at the merge commit, it doesn't include the changes: https://github.com/confluentinc/ksql/blob/a06585d285f4693963c7618c117133102cd655a7/ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java

This also true for commits like #1816 (which happened just before) where the changes haven't made it to master, and haven't been reflected in our docs: https://docs.confluent.io/current/ksql/docs/developer-guide/syntax-reference.html#create-table

But this is not true for commits like #1783 (which happened just after): in this case all the merges happened cleanly and if you checkout master at the merge commit, it includes the changes.

This means something was wacky in the process of merging the release branch into master in some specific cases.

**Comment by mbrancato** [comment]
Any update on this @hjafarpour ? I noticed 5.0.1 was released, Do you think these merge issues will be fixed for 5.1.0 final release?

**Comment by apurvam** [comment]
@mbrancato It is in 5.1.x now: https://github.com/confluentinc/ksql/pull/2122
